I2C_eeprom.cpp

```diff
@@ -306,31 +306,81 @@ uint32_t I2C_eeprom::determineSize(const bool debug)
 
 
 //  new 1.8.1 #61
+//
+// Returns:
+//  0 if device size cannot be determined or device is not online
+//  1 if device has default bytes in first dataFirstBytes bytes [0-BUFSIZE]
+//      Write some dataFirstBytes to the first bytes and retry or use the determineSize method
+//  2 if device has all the same bytes in first dataFirstBytes bytes [0-BUFSIZE]
+//      Write some random dataFirstBytes to the first bytes and retry or use the determineSize method
+//  >= 128 Device size in bytes
 uint32_t I2C_eeprom::determineSizeNoWrite()
 {
+  #define BUFSIZE (32)
   //  try to read a byte to see if connected
   if (!isConnected()) return 0;
 
   bool addressSize = _isAddressSizeTwoWords;
-  byte dummyVal = 0;
-  uint32_t lastOkSize = 0;
+  _isAddressSizeTwoWords = true; //Otherwise reading large EEPROMS fails
+  bool isModifiedFirstSector = false;
+  bool dataIsDifferent = false;
 
-  for (uint32_t size = 128; size <= 65536; size *= 2)
+  byte dataFirstBytes[BUFSIZE];
+  byte dataMatch[BUFSIZE];
+  readBlock(0, dataFirstBytes, BUFSIZE);
+
+  for (uint8_t pos = 0; pos < BUFSIZE; pos++)
```

Comment:
loop can exit if one of the two flags is set.

Reply:
Only if both flags are set